ts/main.ts

```diff
@@ -199,3 +226,108 @@ Make the answers more robust, but not super technical.
     throw new Error(`Error Fetching AI Response: ${error}`);
   }
 }
+
+// Study Mode
+
+const $studyModeBtn = document.getElementById(
+  'study-mode-btn',
+) as HTMLButtonElement;
+$studyModeBtn.addEventListener('click', enterStudyMode);
+
+function enterStudyMode(): any {
+  setActiveNavButton('study-mode-btn');
+  document.querySelector('.input-container')?.classList.add('hidden');
+  document.querySelector('.flashcard-gen-box')?.classList.add('hidden');
+  document.querySelector('#study-mode-container')?.classList.remove('hidden');
+
+  loadFlashcards();
+  displayFlashcard(0);
+}
+
+// Exit Study Mode
+document.getElementById('generate-btn')?.addEventListener('click', () => {
+  setActiveNavButton('generate-btn');
+  document.querySelector('.input-container')?.classList.remove('hidden');
+  document.querySelector('.flashcard-gen-box')?.classList.remove('hidden');
+  document.querySelector('#study-mode-container')?.classList.add('hidden');
```

Comment:
You are querying for a lot of the same things over and over again. Rather than repeating query selectors, you should just query for the element once in the global scope so that way you always have a reference to it. You want to avoid doing querySelectors inside of functions (especially events) because DOM operations are very costly and you are repeating the search more than is needed